packages/server/src/routes/provider/resource/draft/[_id].svelte

```diff
@@ -2,56 +2,53 @@
   import { ResourceSchedule } from "@upswyng/common";
 
   export async function preload({ params, query }, { user }) {
-    if (!user || !user.isAdmin) {
-      this.error(401, "You must be an admin to access this page.");
+    let draftResource = null;
+    if (!user.isAdmin) {
+      // If user is not an admin, make sure the draft they're trying to view is
+      // one of their own
+      const draftsForUserResponse = await this.fetch("/api/resources/drafts/mine", {
+        credentials: "same-origin",
+      });
+      if (draftsForUserResponse.status !== 200) {
+        this.error(draftsForUserResponse.status, draftsForUserResponse.message);
+      }
+      const {draftResources} = await draftsForUserResponse.json();
+      draftResource = draftResources.find(draft => draft._id === params._id);
+      if(!draftResource) {
+        this.error(401, "You don't have permission to view this page");
+      } 
```

Comment:
Maybe we should just move this to the bottom of this function and supply a generic, "There was a problem locating this draft." message. Thoughts?

Reply:
If we don't care about checking to make sure the `draft_.id` is within the users' own drafts, lines 5-28 could be massively simplified to just
```js
    const draftResourceResponse = await this.fetch(
      `/api/resource/draft/${params._id}`
    );
    if (draftResourceResponse.status !== 200) {
      this.error(draftResourceResponse.status, "There was a problem locating this draft.");
    }
    const { draftResource } = await draftResourceResponse.json();
```
for both admins and non-admins.

If we want to stick with checking a users' drafts and making sure they're accessing one of their own drafts, then I think the specificity of the 401 is nice to communicate. (Will admit that the check is a bit YAGNI as I was thinking ahead to this page allowing users to modify their draft, so I'd be fine with ditching it for now in the service of code clarity).

Reply:
> If we want to stick with checking a users' drafts and making sure they're accessing one of their own drafts, then I think the specificity of the 401 is nice to communicate.

Ok I can get on board with that 👍  I think it's alright to surface the drafts existence but explain that they can't access this.